services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -1,25 +1,70 @@
 #pylint: disable=C0111
+import asyncio
 import enum
 import json
 import logging
 from typing import Dict, List, Tuple
 
 import aiohttp
+from aiohttp import web
 from yarl import URL
 
 from . import config, exceptions
 
 DEPENDENCIES_LABEL_KEY = 'simcore.service.dependencies'
 
-NUMBER_OF_RETRIEVED_REPOS = 5
-NUMBER_OF_RETRIEVED_TAGS = 5
+NUMBER_OF_RETRIEVED_REPOS = 50
+NUMBER_OF_RETRIEVED_TAGS = 50
 
 _logger = logging.getLogger(__name__)
+_registry_requests_cache = {}
```

Comment:
Avoid singleton for registry. Add it instead in the app config SEE https://aiohttp.readthedocs.io/en/stable/web_advanced.html#data-sharing-aka-no-singletons-please

Reply:
this was painful... but I think I got it right __and__ with a decorator...
all tests are modified accordingly...